### PR TITLE
Accept a fixed set of allowed formats for timestamps in parsers

### PR DESCRIPTION
**Pull request by spenced**

### CODE REVIEW

File: src/vse_sync_pp/parsers/parser.py [hunk 1]
```diff
@@ -3,7 +3,46 @@
...
+
+    Accepted absolute timestamp strings are ISO 8601 format with restrictions.
+    The string must: explicitly specify UTC timezone, supply time in seconds,
+    specify a decimal fractional part with a decimal mark of '.'.
+
+    Return None if `val` is not a string or is not an ISO 8601 format string.
+    Raise :class:`ValueError` otherwise.
+    """
+    try:
+        dtv = datetime.fromisoformat(val)
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/parser.py:26]: That one is not working for python 3.10. Waiting for update or add as requirement 3.11 (wfm).

File: src/vse_sync_pp/parsers/parser.py [hunk 2]
```diff
@@ -24,14 +24,25 @@ def parse_timestamp_abs(val):
...
         return None
+    except ValueError:
+        # before Python 3.11 `fromisoformat` fails if UTC denoted by 'Z' and/or
+        # the decimal fraction has more than 6 digits
+        match = RE_ISO8601_DECFRAC.match(val)
+        if match is None:
+            return None
+        # parse without decimal fraction, with 'Z' substituted
+        dtv = datetime.fromisoformat(
+            val.group(1) + '+00:00' if val.group(3) == 'Z' else val.group(3)
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/parser.py:37]: ` match.group(1) + '+00:00' if match.group(3) == 'Z' else match.group(3)`

File: src/vse_sync_pp/parsers/parser.py [hunk 3]
```diff
@@ -3,7 +3,57 @@
...
+        return None
+    except ValueError:
+        # before Python 3.11 `fromisoformat` fails if UTC denoted by 'Z' and/or
+        # the decimal fraction has more than 6 digits
+        match = RE_ISO8601_DECFRAC.match(val)
+        if match is None:
+            return None
+        # parse without decimal fraction, with 'Z' substituted
+        dtv = datetime.fromisoformat(
+            val.group(1) + '+00:00' if val.group(3) == 'Z' else val.group(3)
```
Comment (jnunyez) [inline src/vse_sync_pp/parsers/parser.py:37]:
```suggestion
            match.group(1) + '+00:00' if match.group(3) == 'Z' else match.group(3)
```

**Review by jnunyez — APPROVED** [review]

File: src/vse_sync_pp/parsers/parser.py [hunk 4]
```diff
@@ -3,7 +3,57 @@
...
+        dtv = datetime.fromisoformat(
+            match.group(1) + '+00:00' if match.group(3) == 'Z' else match.group(3)
+        )
+    else:
+        match = RE_ISO8601_DECFRAC.match(val)
+        if match is None:
+            raise ValueError(val)
+    if dtv.tzinfo != timezone.utc:
+        raise ValueError(val)
+    return Decimal(f'{int(dtv.timestamp())}.{match.group(2)}')
```
Comment (nocturnalastro) [inline src/vse_sync_pp/parsers/parser.py:45]: A comment about the seconds getting truncated here would be helpful.

Comment (spenced) [reply src/vse_sync_pp/parsers/parser.py:45]: Happy to oblige

**Review by nocturnalastro — APPROVED** [review]

**Pull request merged**